amgut/lib/data_access/sql_connection.py

```diff
@@ -53,6 +53,7 @@ def __init__(self):
         self._connection = None
         self._post_commit_funcs = []
         self._post_rollback_funcs = []
+        self.auto_rollback = False
```

Comment:
Ok, how is that handled in other parts of the test? There are no tests that are modifying the database?

As an alternative, you can add a new element that you are going to delete in the test.